stacks-common/src/address/c32.rs

```diff
@@ -18,12 +18,40 @@ use super::Error;
 
 use sha2::Digest;
 use sha2::Sha256;
-
-const C32_CHARACTERS: &str = "0123456789ABCDEFGHJKMNPQRSTVWXYZ";
+use std::convert::TryFrom;
+
+const C32_CHARACTERS: &[u8; 32] = b"0123456789ABCDEFGHJKMNPQRSTVWXYZ";
+
+/// C32 chars as an array, indexed by their ASCII code for O(1) lookups.
+/// Supports lookups by uppercase, lowercase, and special (i.e. `O, L, I`) chars.
+/// Table can be generated with:
+/// ```
+/// let mut table: [isize; 128] = [-1; 128];
+/// let alphabet = "0123456789ABCDEFGHJKMNPQRSTVWXYZ";
+/// for (i, x) in alphabet.as_bytes().iter().enumerate() {
+///     table[*x as usize] = i as isize;
+/// }
+/// let alphabet_lower = alphabet.to_lowercase();
+/// for (i, x) in alphabet_lower.as_bytes().iter().enumerate() {
+///     table[*x as usize] = i as isize;
+/// }
+/// let specials = [('O', '0'), ('L', '1'), ('I', '1')];
+/// for pair in specials {
+///     let i = alphabet.find(|a| a == pair.1).unwrap() as isize;
+///     table[pair.0 as usize] = i;
+///     table[pair.0.to_ascii_lowercase() as usize] = i;
+/// }
+/// ```
+const C32_CHARACTERS_MAP: [i8; 128] = [
```

Comment:
I experimented with trying both lazy_static and `const fn` for this, neither seemed right, so I took the table result and inlined it. The rust code doc has compilation validation support (at least in the tools I'm using). So seems like this is the best approach. But I'm open to alternatives!